src/base/OpenMCCellAverageProblem.C

```diff
@@ -652,14 +671,10 @@ OpenMCCellAverageProblem::initialSetup()
     if (!_skinner)
       paramError("skinner", "The 'skinner' user object must be of type MoabSkinner!");
 
-    // If the density bins are > 1, we need to re-init the OpenMC materials once (between
-    // the first time we read the materials.xml and when we run the first skinned OpenMC
-    // geometry) so that we have unique materials that can be set to individual densities.
-    // This is just not yet implemented. I think it'd be worthwhile to just allow OpenMC
-    // to set densities of cells filled by the same material, i.e. have the "density"
-    // construct attached to the cell, not material.
-    if (_skinner->nDensityBins() > 1)
-      paramError("skinner", "Density binning is not currently supported for the OpenMC wrapping!");
+    if (_skinner->hasDensitySkinning() != _has_fluid_blocks)
```

Comment:
Can you elaborate on the "vice versa" piece of this message for my benefit?

Reply:
I guess this isn't quite clear to me (though maybe it should be). I know this will only appear if the skinning method is being used, but it reads to me as though you must use the skinner to get fluid feedback in fluid blocks w/ Cardinal, which isn't the case right?

Reply:
This is saying that, if the user specified that their whole OpenMC model should only receive temperature feedback (and not density feedback), that we should throw an error if they are trying to apply density skinning. It is incongruous with the settings they specified. In one place in the input file, they are saying they want to apply density feedback, but in another place they are saying they don't want to apply density feedback.

Reply:
Okay, thanks for clarifying! Same page now.